### PR TITLE
Retire LWI met-field

### PR DESCRIPTION
**Pull request by lizziel**
**This PR is 1 of 2 for super-projects. See instructions here:**
https://github.com/geoschem/GCHP/issues/264
https://github.com/geoschem/GCClassic/issues/26

This pull request retires use of input meteorology field land-water-ice (LWI) index and the State_Met array it is stored in (`State_Met%LWI`). We are removing it as a dependency because it will not be included in the GEOS-IT dataset and determining whether a grid cell is over ocean, ice, or land is possible through other input meteorology fields. It is currently only used for computation in (1) dry deposition if using optional Luo wet deposition, and (2) HEMCO extensions.

Note that the LWI input field is currently computed during GEOS-Chem input pre-processing from GMAO meteorology constants FRLAND, FRLANDIC, FRLAKE, and FROCEAN, and time-varying fields FRSEAICE and FRSNO. Changing to using these fields at run-time will result in differences in the model. In addition, the new way for determining if mostly land, ice, or ocean evaluates the maximum fraction of land (including lake), ice (over both land and ocean), and ocean, which is slightly different than the algorithm for computing LWI. This will also cause differences.

HEMCO PR https://github.com/geoschem/HEMCO/pull/170 should be merged at the same time as this PR.

Closes https://github.com/geoschem/geos-chem/issues/1413

### CODE REVIEW

**Comment by lizziel** [comment]
Results of a 1-month benchmark comarison with and without this update show differences in the land/ocean/ice boundaries, as shown here:
Screen Shot 2022-10-17 at 3 57 03 PM
Screen Shot 2022-10-17 at 3 56 51 PM

**Comment by jimmielin** [comment]
Thanks for this PR. Tagging @fengx7. We will have to make corresponding changes in WRF-GC as well.

File: Interfaces/GEOS/geos_interface.F90 [hunk 1]
```diff
@@ -1271,10 +1270,18 @@ SUBROUTINE GEOS_Diagnostics( GC, IMPORT,  EXPORT, Clock, Phase, &
...
+                + State_Met%FRLAKE(I,J),                          &
+                State_Met%FRSEAICE(I,J),                          &
+                State_Met%FROCEAN(I,J) - State_Met%FRSEAICE(I,J)  &
+             /) )
+
              ! If there are HEMCO lightning emissions in current grid box set
              ! lightning potential accordingly
              IF ( ASSOCIATED(PtrEmis) ) THEN
-                IF ( LWI(I,J) == 1 ) THEN
+                IF ( SfcTypeIndex(1) == 1 ) THEN
```
Comment (msulprizio) [inline Interfaces/GEOS/geos_interface.F90:1284]: Do we need the `(1)` here? It looks like `SfcTypeIndex` is a scalar

File: Interfaces/GEOS/geos_interface.F90 [hunk 2]
```diff
@@ -1284,7 +1291,7 @@ SUBROUTINE GEOS_Diagnostics( GC, IMPORT,  EXPORT, Clock, Phase, &
 
              ! Lightning flash rate
              IF ( LFR(I,J) > 0.0 ) THEN
-                IF ( LWI(I,J) == 1 ) THEN
+                IF ( SfcTypeIndex(1) == 1 ) THEN
```
Comment (msulprizio) [inline Interfaces/GEOS/geos_interface.F90:1294]: Same question here

**Review by msulprizio — COMMENTED** [review]
@lizziel These updates look good to merge. I just had one question denoted here.

File: Interfaces/GEOS/geos_interface.F90 (hunk 2 above)
Comment (yantosca) [reply Interfaces/GEOS/geos_interface.F90:1294]: I think `SfcTypeIndex(1)` can just be reduced to `SfcTypeIndex`.  Functionally they are the same.

Comment (msulprizio) [reply Interfaces/GEOS/geos_interface.F90:1294]: Thanks @yantosca. If there are no objections from @lizziel then I can make this change before merging.

Comment (lizziel) [reply Interfaces/GEOS/geos_interface.F90:1294]: I think that's okay. I originally had it as vector. If it ends up being a problem I can take care of it when I test GEOS.

Comment (msulprizio) [reply Interfaces/GEOS/geos_interface.F90:1294]: Thanks for confirming!

**Pull request merged**